[Tidy; Docs] Remove kedro _get_dataset call; enhance kedro docs

antonymilne opened the pull request:
## Description

* Remove the `_get_dataset` call (only done for `kedro>=0.19.9` even though we could maybe do it for some earlier versions. So the legacy code will remain for some time, but in future we now have a much better way to handle the data catalog that doesn't require any hacking 🎉 Thanks kedro team!)
* Update data manager docs to fully explain the use of `catalog.yaml` as a data source registry _outside a kedro project_ and expand on some other details that weren't explained before (I hope this is still intended use of `DataCatalog` @astrojuanlu? It was never very commonly done even though kedro had always designed it to be used like this)

FYI @astrojuanlu @gtauzin

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

    - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
    - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
    - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
    - I have not referenced individuals, products or companies in any commits, directly or indirectly.
    - I have not added data or restricted code in any commits, directly or indirectly.

antonymilne commented on vizro-core/docs/pages/user-guides/kedro-data-catalog.md:
```diff
@@ -10,67 +12,118 @@ If you already have Kedro installed then you do not need to install any extra de
 
 !!! example "Import a Kedro Data Catalog into the Vizro data manager"
-    === "app.py (Kedro project path)"
+    === "app.py (Data Catalog configuration file)"
```
This is just switching the order of examples to match the new above ordering, no code changes.

antonymilne commented on vizro-core/src/vizro/integrations/kedro/_data_manager.py:
```diff
@@ -65,9 +65,6 @@ def datasets_from_catalog(catalog: CatalogProtocol, *, pipeline: Pipeline = None
-            #  but need to check if works with caching.
-            dataset = catalog._get_dataset(dataset_name, suggest=False)
-            vizro_data_sources[dataset_name] = dataset.load
+            vizro_data_sources[dataset_name] = lambda: catalog.load(dataset_name)
```
This is the major code improvement 🎉

gtauzin commented on vizro-core/docs/pages/user-guides/kedro-data-catalog.md:
```diff
@@ -1,6 +1,8 @@
-This page describes how to integrate Vizro with [Kedro](https://docs.kedro.org/en/stable/index.html), an open-source Python framework to create reproducible, maintainable, and modular data science code. For Pandas datasets registered in a Kedro data catalog, Vizro provides a convenient way to visualize them.
+This page describes how to integrate Vizro with [Kedro](https://docs.kedro.org/en/stable/index.html), an open-source Python framework to create reproducible, maintainable, and modular data science code. For Pandas datasets registered in a [Kedro Data Catalog](https://docs.kedro.org/en/stable/data/index.html), Vizro provides a convenient way to visualize them.
+
+Even if you do not have a Kedro project, you can still [use a Kedro Data Catalog](#create-a-kedro-data-catalog) to manage your dashboard's data sources.
```
I find this super nice. :100: 

I wonder how hard it is to do if you also want to use credentials (e.g. to access data on s3 buckets)

antonymilne replied:
Yeah, my first draft actually included credentials as well but then I decided not to include it, but maybe I should do so. Basically the simplest approach would be to do this:
```
catalog_yaml = yaml.safe_load(Path("catalog.yaml").read_text(encoding="utf-8"))
credentials_yaml = yaml.safe_load(Path("catalog.yaml").read_text(encoding="utf-8"))
DataCatalog.from_config(catalog_yaml, credentials_yaml)
```

The more advanced kedro way to do it is to use `ConfigLoader` rather than just loading with `yaml.safe_load`, [like in this example](https://docs.kedro.org/en/stable/configuration/configuration_basics.html#how-to-load-a-data-catalog-with-credentials-in-code). But my feeling is that not many vizro users would want to do that - either you don't have a kedro project, in which case using a config loader with different run environments etc. is overkill, or you already have a kedro project, in which case you're not trying to manually load the catalog from file anyway (you use `catalog_from_project` instead).

But just now I actually noticed you can use `OmegaConfigLoader` without need the whole `conf` structure, so maybe that's a better way to do it 🤔 https://docs.kedro.org/en/stable/configuration/advanced_configuration.html#advanced-configuration-without-a-full-kedro-project

gtauzin replied:
`OmegaConfigLoader` makes very neat!

antonymilne replied:
Done in [ef6db1c](https://github.com/mckinsey/vizro/pull/1014/commits/ef6db1cf6f28bb075314ae82d4f7ca7d6e044dce)!

stichbury commented on vizro-core/docs/pages/user-guides/data.md:
```diff
@@ -38,7 +38,19 @@ graph TD
     | Production-ready                                              | Yes              | Yes                                      |
 
-If you have a [Kedro](https://kedro.org/) project or would like to use the [Kedro Data Catalog](https://docs.kedro.org/en/stable/data/index.html) to manage your data independently of a Kedro project then you should use Vizro's [integration with the Kedro Data Catalog](kedro-data-catalog.md). This offers helper functions to add [`kedro_datasets.pandas`](https://docs.kedro.org/en/stable/kedro_datas ...
+If you have a [Kedro](https://kedro.org/) project then you should use Vizro's [integration with the Kedro Data Catalog](kedro-data-catalog.md) to add [`kedro_datasets.pandas`](https://docs.kedro.org/en/stable/kedro_datasets.html) data to the Vizro data manager.
```
```suggestion
If you have a [Kedro](https://kedro.org/) project, you should use Vizro's [integration with the Kedro Data Catalog](kedro-data-catalog.md) to add [`kedro_datasets.pandas`](https://docs.kedro.org/en/stable/kedro_datasets.html) data to the Vizro data manager.
```

stichbury commented on vizro-core/docs/pages/user-guides/kedro-data-catalog.md:
```diff
@@ -1,6 +1,8 @@
+# How to integrate Vizro with the Kedro Data Catalog
 
-This page describes how to integrate Vizro with [Kedro](https://docs.kedro.org/en/stable/index.html), an open-source Python framework to create reproducible, maintainable, and modular data science code. For Pandas datasets registered in a Kedro data catalog, Vizro provides a convenient way to visualize them.
+This page describes how to integrate Vizro with [Kedro](https://docs.kedro.org/en/stable/index.html), an open-source Python framework to create reproducible, maintainable, and modular data science code. For Pandas datasets registered in a [Kedro Data Catalog](https://docs.kedro.org/en/stable/data/index.html), Vizro provides a convenient way to visualize them.
```
```suggestion
This page describes how to integrate Vizro with [Kedro](https://docs.kedro.org/en/stable/index.html), an open-source Python framework to create reproducible, maintainable, and modular data science code. Vizro provides a convenient way to visualize Pandas datasets registered in a [Kedro Data Catalog](https://docs.kedro.org/en/stable/data/index.html).
```

stichbury commented on vizro-core/docs/pages/user-guides/kedro-data-catalog.md:
```diff
@@ -10,63 +12,111 @@ If you already have Kedro installed then you do not need to install any extra de
+
+## Create a Kedro Data Catalog
+
+You can create a [Kedro Data Catalog](https://docs.kedro.org/en/stable/data/index.html) to be a YAML registry of your dashboard's data sources. If you have a Kedro project then you will already have this file; if you would like to use the Kedro Data Catalog outside a Kedro project then you need to create it. In this case, you should save your `catalog.yaml` file to the same directory as your `app ...
```
I feel like we should have more instructions on what it means to create it (either a link to instructions or just say explicitly that you need to create a yaml file and where to put it, if the instructions for writing it are below).

stichbury commented on vizro-core/docs/pages/user-guides/kedro-data-catalog.md:
```diff
@@ -10,63 +12,111 @@ If you already have Kedro installed then you do not need to install any extra de
+
+1. The [minimum details needed](https://docs.kedro.org/en/stable/data/data_catalog.html#the-basics-of-catalog-yml) for a Kedro Data Catalog entry are the data source name (`cars`), the type of data (`type`), and the file's location (`filepath`).
+1. Vizro supports all [`kedro_datasets.pandas`](https://docs.kedro.org/en/stable/kedro_datasets.html) datasets. This includes, for example, CSV, Excel and Parquet files.
+1. Kedro supports a [variety of data stores](https://docs.kedro.org/en/stable/data/data_catalog.html#dataset-filepath) including local file systems, network file systems and cloud object store.
```
```suggestion
1. Kedro supports a [variety of data stores](https://docs.kedro.org/en/stable/data/data_catalog.html#dataset-filepath) including local file systems, network file systems and cloud object stores.
```

stichbury commented on vizro-core/docs/pages/user-guides/kedro-data-catalog.md:
```diff
@@ -10,63 +12,111 @@ If you already have Kedro installed then you do not need to install any extra de
 ## Use datasets from the Kedro Data Catalog
 
-`vizro.integrations.kedro` provides functions to help generate and process a [Kedro Data Catalog](https://docs.kedro.org/en/stable/data/index.html). It supports both the original [`DataCatalog`](https://docs.kedro.org/en/stable/data/data_catalog.html) and the more recently introduced [`KedroDataCatalog`](https://docs.kedro.org/en/stable/data/index.html#kedrodatacatalog-experimental-feature). Give ...
+Vizro provides functions to help generate and process a [Kedro Data Catalog](https://docs.kedro.org/en/stable/data/index.html) in the module `vizro.integrations.kedro`. This supports both the original [`DataCatalog`](https://docs.kedro.org/en/stable/data/data_catalog.html) and the more recently introduced [`KedroDataCatalog`](https://docs.kedro.org/en/stable/data/index.html#kedrodatacatalog-exper ...
```
```suggestion
Vizro provides functions to help generate and process a [Kedro Data Catalog](https://docs.kedro.org/en/stable/data/index.html) in the module `vizro.integrations.kedro`. The module supports both the original [`DataCatalog`](https://docs.kedro.org/en/stable/data/data_catalog.html) and the more recently introduced [`KedroDataCatalog`](https://docs.kedro.org/en/stable/data/index.html#kedrodatacatalog-experimental-feature). Given a Kedro `catalog`, the general pattern to add datasets to the Vizro data manager is:
```

stichbury commented on vizro-core/docs/pages/user-guides/kedro-data-catalog.md:
````diff
@@ -10,63 +12,111 @@ If you already have Kedro installed then you do not need to install any extra de
 ```
 
-This imports all datasets of type [`kedro_datasets.pandas`](https://docs.kedro.org/en/stable/kedro_datasets.html) from the Kedro `catalog` into the Vizro `data_manager`.
+This registers in the Vizro `data_manager` all data sources of type [`kedro_datasets.pandas`](https://docs.kedro.org/en/stable/kedro_datasets.html) from the Kedro `catalog`. You can now [reference the data source](data.md#reference-by-name) by name. For example, given the [above `catalog.yaml` file](#create-a-kedro-data-catalog), you could use the data source names `"cars"`, `"motorbikes"`, `"tra ...
````
```suggestion
The code above registers all data sources of type [`kedro_datasets.pandas`](https://docs.kedro.org/en/stable/kedro_datasets.html) from the Kedro `catalog` with Vizro's `data_manager` . You can now [reference the data source](data.md#reference-by-name) by name. For example, given the [above `catalog.yaml` file](#create-a-kedro-data-catalog), you could use the data source names `"cars"`, `"motorbikes"`, `"trains"`, and `"trucks"` with `px.scatter("cars", ...)`.
```

stichbury commented on vizro-core/docs/pages/user-guides/kedro-data-catalog.md:
```diff
@@ -10,63 +12,111 @@ If you already have Kedro installed then you do not need to install any extra de
+This registers in the Vizro `data_manager` all data sources of type [`kedro_datasets.pandas`](https://docs.kedro.org/en/stable/kedro_datasets.html) from the Kedro `catalog`. You can now [reference the data source](data.md#reference-by-name) by name. For example, given the [above `catalog.yaml` file](#create-a-kedro-data-catalog), you could use the data source names `"cars"`, `"motorbikes"`, `"tra ...
+
+!!! note
+    Data sources imported from Kedro in this way are [dynamic data](data.md#dynamic-data). This means that the data can be refreshed while your dashboard is running. For example, if you execute a Kedro pipeline run then the latest data can be shown in your dashboard without restarting it.
```
```suggestion
    Data sources imported from Kedro in this way are [dynamic data](data.md#dynamic-data). This means that the data can be refreshed while your dashboard is running. For example, if you run a Kedro pipeline, the latest data is shown in the Vizro dashboard without restarting it.
```

stichbury reviewed (approved):
LGTM from a docs perspective. Just a few language tweaks because I thought we could make a complex topic slightly simpler, but really, I don't think any add much, so feel free to ignore.

antonymilne commented:
Thanks for the review @stichbury! I've incorporated your suggestions.

@gtauzin I just updated the docs and our tests to use `OmegaConfigLoader`. Take a look: [ef6db1c](https://github.com/mckinsey/vizro/pull/1014/commits/ef6db1cf6f28bb075314ae82d4f7ca7d6e044dce) and [built docs](https://vizro--1014.org.readthedocs.build/en/1014/pages/user-guides/kedro-data-catalog/).

If you have a chance to look at it and think it looks good then please do try to click the Approve button 🙂

antonymilne commented on vizro-core/src/vizro/integrations/kedro/_data_manager.py:
Reminder to myself to add link to API docs once #1020 merges.

antonymilne replied:
Done in [f1510aa](https://github.com/mckinsey/vizro/pull/1014/commits/f1510aa42d17ffda1daaab2585f20ab6fe9be251).

gtauzin reviewed (approved):
Nice improvements! Really like the way you advocate the use of the kedro catalog to non-kedro users.
